src/pages/home/report/ContextMenu/BaseReportActionContextMenu.tsx

```diff
@@ -135,6 +136,13 @@ function BaseReportActionContextMenu({
         return reportActions[reportActionID];
     }, [reportActions, reportActionID]);
 
+    const message = Array.isArray(reportAction?.message) ? reportAction?.message?.at(-1) ?? null : reportAction?.message ?? null;
+    const html = message?.html ?? '';
+    const {sourceURL} = getAttachmentDetails(html);
+    const sourceID = (sourceURL?.match(CONST.REGEX.ATTACHMENT_ID) ?? [])[1];
```

Comment:
NAB: Can we make util for this?

- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)

Reply:
I updated